src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs

```diff
@@ -0,0 +1,726 @@
+            Assert.True(summarizeResults.Odatacount.HasValue);
+            Assert.True(1 == summarizeResults.Odatacount.Value);
+
+            Assert.NotNull(summarizeResults.Value);
+            Assert.True(1 == summarizeResults.Value.Count);
+
+            var summary = summarizeResults.Value[0];
+
+            Assert.NotNull(summary);
+
+            Assert.Null(summary.Odataid);
+            Assert.False(string.IsNullOrEmpty(summary.Odatacontext));
+
+            Assert.NotNull(summary.Results);
+            Assert.False(string.IsNullOrEmpty(summary.Results.QueryResultsUri));
+            Assert.True(summary.Results.NonCompliantResources.HasValue);
+            Assert.True(summary.Results.NonCompliantPolicies.HasValue);
+
+            Assert.NotNull(summary.PolicyAssignments);
+            Assert.True(summary.PolicyAssignments.Count == summary.Results.NonCompliantPolicies.Value);
+
+            foreach (var policyAssignmentSummary in summary.PolicyAssignments)
+            {
+                Assert.NotNull(policyAssignmentSummary);
+
+                Assert.False(string.IsNullOrEmpty(policyAssignmentSummary.PolicyAssignmentId));
+
+                Assert.NotNull(policyAssignmentSummary.Results);
+                Assert.False(string.IsNullOrEmpty(policyAssignmentSummary.Results.QueryResultsUri));
+                Assert.True(policyAssignmentSummary.Results.NonCompliantResources.HasValue);
+                Assert.True(policyAssignmentSummary.Results.NonCompliantPolicies.HasValue);
+
+                Assert.NotNull(policyAssignmentSummary.PolicyDefinitions);
+                Assert.True(policyAssignmentSummary.PolicyDefinitions.Count == policyAssignmentSummary.Results.NonCompliantPolicies.Value);
+
+                if (policyAssignmentSummary.Results.NonCompliantPolicies.Value > 1)
+                {
+                    Assert.False(string.IsNullOrEmpty(policyAssignmentSummary.PolicySetDefinitionId));
+                }
+
+                foreach (var policyDefinitionSummary in policyAssignmentSummary.PolicyDefinitions)
+                {
+                    Assert.NotNull(policyDefinitionSummary);
+
+                    Assert.False(string.IsNullOrEmpty(policyDefinitionSummary.PolicyDefinitionId));
+                    Assert.False(string.IsNullOrEmpty(policyDefinitionSummary.Effect));
+
+                    Assert.NotNull(policyDefinitionSummary.Results);
+                    Assert.False(string.IsNullOrEmpty(policyDefinitionSummary.Results.QueryResultsUri));
+                    Assert.True(policyDefinitionSummary.Results.NonCompliantResources.HasValue);
+                    Assert.False(policyDefinitionSummary.Results.NonCompliantPolicies.HasValue);
+                }
+            }
+        }
+
+        #endregion
+
+        #region Policy Events - Scopes
+
+        [Fact]
+        public void PolicyEvents_ManagementGroupScope()
+        {
+            var managementGroupName = "AzGovTest1";
```

Comment:
@bulentelmaci does this mean I should be having a pre-created assets e.g. ManagementGroup called AzGovTest1? in order to record this test?

Reply:
Of course. This is an extension RP that relies on real resources to be able to do anything. Management groups, subscription Ids, resource groups, resources, policies, etc.